### ISSUE TITLE
Gas and Ice Giants

### ISSUE THREAD

**Issue by Pin-Lui** [sp00ktober/DSP_Mod#17]
it does not work for Gas and Ice Giants, the collectors plop out after a very short distance. I have the distance all the way up to x10

**Comment by sp00ktober** [comment]
This is because i patched the `GetNearestStarPlanet()` method in such a way that it would return the giants only if there is no other planet nearby. Thought this would make more sense this way as in the vanilla game you have one of them near your home planet and you would rather want to see the factory on your home planet than the few collectors on your gas giant.

Or is the giant you are talking about "on its own"?

**Comment by Pin-Lui** [comment]
No, it has 2 Moons, I use an rtx3090 and 128GB ram so I'm just greedy and I want to see all the things all the time from everywhere xD
if I have to choose Factory > Gas giant. but perfect would be booth xD

edit: btw thx for the mod! the only one i installed

**Comment by sp00ktober** [comment]
Im happy to hear that :)

Both will be hard as the game only wants to render one planet at a time. It _may_ by possible to do it but i'd have to look into it for some time to know.
This would then probably also solve #16
